### PR TITLE
[No Ticket] Add prettierrc

### PR DESCRIPTION
**Pull request by s-rubenstein**
### What
- Adds a prettierrc file for intellij autoformat

### Why
- Otherwise intellij autoformat defaults to double quotes

### Testing strategy
- [ ]

### CODE REVIEW

**Review by nawatts — APPROVED** [review]

**Review by msilva-broad — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Comment by nawatts** [comment]
With this, we can also remove the `singleQuote` option from the `prettier/prettier` rule in ESLint config.
https://github.com/prettier/eslint-plugin-prettier#options

**Comment by s-rubenstein** [comment]
@nawatts does this file work with running eslint as well? I wasn't sure if it would get picked up by `yarn lint`.

I tried running `yarn lint` though, and got:
```
(venv3.7.12) srubenstein@wm154-5d8 terra-ui % yarn lint

/Users/srubenstein/Desktop/code/terra-ui/coverage/lcov-report/block-navigation.js
  0:0  error  Parsing error: ESLint was configured to run on `<tsconfigRootDir>/coverage/lcov-report/block-navigation.js` using `parserOptions.project`: <tsconfigRootDir>/tsconfig.json
However, that TSConfig does not include this file. Either:
- Change ESLint's list of included files to not include this file
- Change that TSConfig to include this file
- Create a new TSConfig that includes this file and include it in your parserOptions.project
See the typescript-eslint docs for more info: https://typescript-eslint.io/linting/troubleshooting#i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file

/Users/srubenstein/Desktop/code/terra-ui/coverage/lcov-report/prettify.js
  0:0  error  Parsing error: ESLint was configured to run on `<tsconfigRootDir>/coverage/lcov-report/prettify.js` using `parserOptions.project`: <tsconfigRootDir>/tsconfig.json
However, that TSConfig does not include this file. Either:
- Change ESLint's list of included files to not include this file
- Change that TSConfig to include this file
- Create a new TSConfig that includes this file and include it in your parserOptions.project
See the typescript-eslint docs for more info: https://typescript-eslint.io/linting/troubleshooting#i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file

/Users/srubenstein/Desktop/code/terra-ui/coverage/lcov-report/sorter.js
  0:0  error  Parsing error: ESLint was configured to run on `<tsconfigRootDir>/coverage/lcov-report/sorter.js` using `parserOptions.project`: <tsconfigRootDir>/tsconfig.json
However, that TSConfig does not include this file. Either:
- Change ESLint's list of included files to not include this file
- Change that TSConfig to include this file
- Create a new TSConfig that includes this file and include it in your parserOptions.project
See the typescript-eslint docs for more info: https://typescript-eslint.io/linting/troubleshooting#i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file

✖ 3 problems (3 errors, 0 warnings)
```

**Comment by nawatts** [comment]
That error is ESLint attempting to lint the unit test coverage report in the `coverage` directory. That report is generated, so we don't care about linting it. #3911 adds the `coverage` directory to `.eslintignore`. You could also delete the `coverage` directory to resolve the error.

File: .eslintrc.js [hunk 1]
```diff
@@ -18,7 +18,6 @@ module.exports = {
     'import/prefer-default-export': 'off',
     'simple-import-sort/imports': 'warn',
     quotes: [2, 'single', { avoidEscape: true }],
-    'prettier/prettier': ['error', { singleQuote: true }],
```
Comment (nawatts) [inline .eslintrc.js:21]: We don't want to remove the rule entirely, just the options. Without options specified, Prettier run via ESLint will use the options configured in `.prettierrc.json`.

```
prettier/prettier': 'error',
```

**Review by nawatts — CHANGES_REQUESTED** [review]

**Review by nawatts — APPROVED** [review]

**Pull request merged**